docker/stan-wasm-server/logic/locking.py

```diff
@@ -0,0 +1,40 @@
+import os
+from random import seed, choices
+from string import ascii_letters
+
+def get_nonce():
+    seed(os.getpid())
+    return str.join('', choices(ascii_letters, k=20))
```

Comment:
since this re-seeds each call, won't sucessive calls from the same thread yield the same nonce?
I guess that's fine, since a thread can't race with itself? 
Or maybe it can once we make more of compilation async. 

Also, and this is super nitpicky, but I've known "nonce" to be short for "number used once", so the fact that it's actually alphabetical feels disingenuous to me somehow

Reply:
Good call on the seed--better to be sure. Maybe `int(str(os.getpid()) + str(int(time.time())))`? (Doubt we'd really need sub-second resolution...)

"Nonce" is used in other contexts for non-numeric data (e.g. linguistics, where it's a one-use sound not otherwise recognized as part of the language) but I'm not attached to it if there's something you'd prefer.

Reply:
is there a particularly compelling reason to not just use 
`int(str(os.getpid()) + str(int(time.time())))`
_as the nonce_?

Reply:
An excellent point